OpenCabConsumer/app/src/main/java/com/eleostech/exampleconsumer/MainActivity.java

```diff
@@ -126,20 +129,20 @@ private void callHOSProvider() {
                                 Log.d(LOG_TAG, "Got result!");
                                 result.setClassLoader(HOSContract.class.getClassLoader());
 
-                                Version maxSupportedVersion = new Version("0.3");
+                                Version maxSupportedVersion = new Version("0.4");
                                 Version resultVersion;
                                 if (result.containsKey(HOSContract.KEY_VERSION) && result.get(HOSContract.KEY_VERSION) != null) {
                                     resultVersion = new Version(result.getString(HOSContract.KEY_VERSION));
                                 } else {
-                                    resultVersion = new Version("0.3");
+                                    resultVersion = new Version("0.4");
                                 }
                                 if (resultVersion.compareTo(maxSupportedVersion) >= 1) {
                                     adapterHos.insert(dateTime + " : " + "Version " + resultVersion + " is not supported", 0);
                                 } else if (result.containsKey(HOSContract.KEY_HOS)) {
-                                    HOSContract.HOSStatusV2 hosStatus = result.getParcelable(HOSContract.KEY_HOS);
+                                    HOSContract.HOSData hosStatus = gson.fromJson(result.getString(HOSContract.KEY_HOS), HOSContract.HOSData.class);
```

Comment:
Doesn't this need to handle V2 responses? (Not all providers will update at the same time.)